io-sim/test/Test/IOSim.hs

```diff
@@ -301,6 +310,34 @@ prop_threadId_order_order_Sim :: Positive Int -> Property
 prop_threadId_order_order_Sim n = runSimOrThrow $ test_threadId_order n
 
 
+prop_wakeup_order_ST :: Property
+prop_wakeup_order_ST = runSimOrThrow $ test_wakeup_order
+
+-- It is not deterministic in IO
+--prop_wakeup_order_IO = ioProperty test_wakeup_order
```

Comment:
Did you leave this behind by mistake or should it remain?

Reply:
It's there to remind us of the fact that this is not in fact deterministic in IO. Arguably since the IO STM doesn't provide this property, we should not provide it in the Sim STM.